a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -478,13 +480,15 @@ class CountLoadbalancersInProjectBySubnet(BaseDatabaseTask):
     def execute(self, lb_resource, subnet):
         try:
             project_id = lb_resource.project_id
-            vthunder_conf = CONF.hardware_thunder.devices[project_id]
-            if (vthunder_conf.hierarchical_multitenancy == 'enable' and
-                    CONF.a10_global.use_parent_partition):
-                partition_name = self.vthunder_repo.get_partition_for_project(
-                    db_apis.get_session(), project_id=project_id)
-                project_ids = self.vthunder_repo.get_project_list_using_partition(
-                    db_apis.get_session(), partition_name=partition_name)
+            vthunder_conf = CONF.hardware_thunder.devices.get(project_id)
+            project_ids = []
+            if vthunder_conf:
```

Comment:
A None value vthunder_conf signifies the configuration for a project is not there in config.  
Is this a valid case or this is an error case?
Checked the existing code where we fetch config inf for project we haven't checked vthunder_config anywhere.

Reply:
yes offcourse it's a valid case. 
We need to consider the case where the `project_id` may not be present in Config file and we handle it accordingly.

Other places that you are talking about, (you can check it though) perhaps, it might be going into the other flow of execution.